build_runner/lib/src/generate/build.dart

```
@@ -92,19 +93,22 @@ Future<BuildResult> build(List<BuilderApplication> builders,
       verbose: verbose,
       buildDirs: buildDirs);
   var terminator = new Terminator(terminateEventStream);
-  var build = await BuildRunner.create(
-    options,
-    environment,
-    builders,
-    builderConfigOverrides,
-    isReleaseBuild: isReleaseBuild ?? false,
-  );
-  if (build == null) return new BuildResult(BuildStatus.failure, []);
-  var result = await build.run({});
-  await build.beforeExit();
-  await terminator.cancel();
-  await options.logListener.cancel();
-  return result;
+  try {
+    var build = await BuildRunner.create(
+      options,
+      environment,
+      builders,
+      builderConfigOverrides,
+      isReleaseBuild: isReleaseBuild ?? false,
+    );
+    var result = await build.run({});
+    await build.beforeExit();
+    await terminator.cancel();
+    await options.logListener.cancel();
+    return result;
+  } on CannotBuildException {
```

I don't think we need to catch this here. We can let it bubble up to `run`